assets/src/hooks/useDetours.ts

```diff
@@ -21,17 +25,25 @@ const subscribe = (
   const channel = socket.channel(topic)
 
   handleDrafted &&
-    channel.on("drafted", ({ data: data }) => handleDrafted(data))
+    channel.on("drafted", ({ data: data }: { data: SimpleDetourData }) =>
+      handleDrafted(simpleDetourFromData(data))
+    )
   handleActivated &&
-    channel.on("activated", ({ data: data }) => handleActivated(data))
+    channel.on("activated", ({ data: data }: { data: SimpleDetourData }) =>
+      handleActivated(simpleDetourFromData(data))
+    )
   handleDeactivated &&
-    channel.on("deactivated", ({ data: data }) => handleDeactivated(data))
+    channel.on("deactivated", ({ data: data }: { data: SimpleDetourData }) =>
+      handleDeactivated(simpleDetourFromData(data))
+    )
   channel.on("auth_expired", reload)
 
   channel
     .join()
-    .receive("ok", ({ data: data }: { data: SimpleDetour[] }) => {
-      const detoursMap = Object.fromEntries(data.map((v) => [v.id, v]))
+    .receive("ok", ({ data: data }: { data: SimpleDetourData[] }) => {
+      const detoursMap = Object.fromEntries(
+        data.map((v) => [v.id, simpleDetourFromData(v)])
+      )
```

Comment:
Example for https://github.com/mbta/skate/pull/2908/files/8acf7072c0392401f45e97dfc4e47d1e83b134a8..ecd135c15c69f5510318f45962d851c35238520e#r1909378792

```suggestion
        .receive("ok", ({ data: unknownData }: { data: unknown }) => {
      const data = create(unknownData, array(SimpleDetourData))

      const detoursMap = Object.fromEntries(
        data.map((v) => [v.id, simpleDetourFromData(v)])
      )
```

Reply:
https://github.com/mbta/skate/pull/2908/commits/69b649c28079d73cc8e1de3e8bb0e84fcabd46a5